src/mixer/basetrackplayer.cpp

```diff
@@ -245,6 +258,18 @@ void BaseTrackPlayerImpl::disconnectLoadedTrack() {
 }
 
 void BaseTrackPlayerImpl::slotLoadTrack(TrackPointer pNewTrack, bool bPlay) {
+    mixxx::Duration elapsed = m_cloneTimer.restart();
+    if (elapsed < mixxx::Duration::fromSeconds(1)) {
+        // load pressed twice quickly, clone instead of loading
+        EngineChannel* pChannel = m_pEngineMaster->getEngineSync()->pickNonSyncSyncTarget(m_pChannel);
```

Comment:
this needs to be replaced with something more specific and predictable, but I don't know what that is. I'll need some guidance here.